src/App.tsx

```diff
@@ -1,14 +1,88 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [iconStates, setIconStates] = useState<{ [key: number]: boolean }>({});
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+      try {
+        const data = await getTodos();
+
+        setTodos(data);
+      } catch (error) {
+        throw new Error('Error fetching todos');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  const handleSelectTodo = async (todo: Todo) => {
+    setIsLoading(true);
+    setSelectedTodo(todo);
+    setSelectedUser(null);
+    setIconStates(prevState => ({
+      ...prevState,
+      [todo.id]: true,
+    }));
+    try {
+      const user = await getUser(todo.userId);
+
+      setSelectedUser(user);
+    } catch (error) {
+      throw new Error('Error fetching user');
```

Comment:
Similar to the previous comment, instead of throwing a new error, consider logging the error or displaying a user-friendly message to improve user experience.